src/query/functions/src/scalars/timestamp/src/datetime.rs

```diff
@@ -502,7 +537,37 @@ fn register_number_to_timestamp(registry: &mut FunctionRegistry) {
 fn register_string_to_date(registry: &mut FunctionRegistry) {
     registry.register_passthrough_nullable_1_arg::<StringType, DateType, _, _>(
         "to_date",
-        |_, _| FunctionDomain::MayThrow,
+        |ctx, d| {
```

Comment:
Here just check, the maximum and the minimum, what happens if the value is_err in the middle?

Reply:
I think it will throw errors during runtime ( to_date will process all values of this column).

So it maybe ok to include it in domain calculation?

Reply:
Yes. If is_err return maythrow domain. Real err check in eval logic